src/libraries/System.Private.CoreLib/src/System/Runtime/InteropServices/ComWrappers.cs

```diff
@@ -314,15 +314,22 @@ public int QueryInterface(in Guid riid, out IntPtr ppvObject)
                         }
                         else
                         {
-                            Guid riidLocal = riid;
-                            switch (customQueryInterface.GetInterface(ref riidLocal, out ppvObject))
+                            try
                             {
-                                case CustomQueryInterfaceResult.Handled:
-                                    return HResults.S_OK;
-                                case CustomQueryInterfaceResult.NotHandled:
-                                    break;
-                                case CustomQueryInterfaceResult.Failed:
-                                    return HResults.COR_E_INVALIDCAST;
+                                Guid riidLocal = riid;
+                                switch (customQueryInterface.GetInterface(ref riidLocal, out ppvObject))
```

Comment:
Can we do `Unsafe.AsRef` on the RIID here to avoid the local copy? AFAIK this takes `ref` here only for historical reasons, but it's not valid to mutate the IID anyway.